app/models/search.server.ts

```diff
@@ -42,6 +42,23 @@ function calculateElementsRemaining(remaining: number, gotten: number) {
   return remaining - gotten
 }
 
+export async function getSearchResults(
+  text: string,
+  page: string = "1",
+  caseSensitive: boolean = false,
+  attribute: string = SearchResultEnum.HEADWORD
+) {
+  switch (attribute) {
+    case SearchResultEnum.HEADWORD:
+      return getEntriesByBasicTextSearchAndPage(text, page, caseSensitive)
+    case SearchResultEnum.MEANING:
+      return getSearchResultsFromMeaningsAndPage(text, page, caseSensitive)
+    default:
+      throw new Error(`attribute ${attribute} must be a valid search result`)
+  }
+}
```

Comment:
So I was thinking pertaining to this (and to the other switch statement on Enum):

Instead we could make some typescript classes and have them handle this behaviour. This means that the server code won't be handled in files like this, but I think it could make the code maintainable. Rather than Having this switch statement, we could then call `Object.GetSearchResults(text, page, caseSensitive)` and just return that.

This would also apply to the other Switch Statement in `SearchResults.tsx`. We could replace that with `GetJsx()` on some SearchResultObject.

Thoughts on this? Downsides would be:
1. it's a bit of work, would probably take several hours to refactor so I don't know if it's worth it
2. the code for the server stuff would be scattered between the objects. I think this is fine but it definitely goes against the current "style" of how things are right now. We don't use much inheritance atm and I'm not sure how that fits with the typescript philosophy. It's probably how I'd solve this problem if I was using C# though.

Honestly I really don't mind keeping things as they are-- I think it's chill and not a big deal and the code is pretty readable and straightforward. Just wanted to bring this up in case you thought it was a good idea

Reply:
Let's leave it as it is. A class-oriented approach could make sense, too, but for now I think this is fine.